boards/arm/frdm_k22f/frdm_k22f.dts

```diff
@@ -174,6 +174,12 @@ zephyr_udc0: &usbotg {
 	num-bidir-endpoints = <8>;
 };
 
+zephyr_uhc0: &uhc1 {
```

Comment:
We should not need to set the compatible or controller index here at the board level, these are SOC level properties. Same comments apply to other SOCs enabled in this PR.

Also, we do have a K22 SOC enabled in tree with a SYSMPU instance: https://github.com/zephyrproject-rtos/zephyr/blob/daf7c80c1599bcce969877b7d451e66496cf28e5/dts/arm/nxp/nxp_k22fx512.dtsi#L1. We should add the SYSMPU setup code to the K22 init if this SOC Kconfig is set (`CONFIG_SOC_MK22F12`)